src/components/Main.tsx

```diff
@@ -63,11 +66,16 @@ export function Main({
   /** Set of supported languages */
   supportedLanguages: ConsentManagerLanguageKey[];
 }): JSX.Element {
-  // need to focus the first button in the modal when the modal is opened
+  const { formatMessage } = useIntl();
+  // need to focus the element marked with data-initialFocus when the modal is opened
+  // regular autofocus attributes caused errors, thus the data attribute usage
+  // NOTE: if we want to meet a11y guidelines we will need to implement a focus trap as well
   const dialogRef = useRef<HTMLDivElement>(null);
   useEffect(() => {
     if (!isViewStateClosed(viewState) && dialogRef.current) {
-      dialogRef?.current?.querySelector('button')?.focus();
+      setTimeout(() => {
```

Comment:
Because `setTimeout`s are often kind of "hacky" it could be a good idea to leave a comment to explain why this is necessary.